src/core/views.py

```diff
@@ -3559,14 +3714,143 @@ def view_proposal_review(request, proposal_id, assignment_id, access_key=None):
         'recommendation_form': recommendation_form,
         'book_editors': _proposal.book_editors.all(),
         'active': 'proposal_review',
-        'instructions': get_setting('instructions_for_task_proposal',
-                                    'general'),
+        'instructions': get_setting(
+            'instructions_for_task_proposal',
+            'general'
+        ),
         'data': data,
     }
 
     return render(request, template, context)
 
 
+class ProposalReviewCompletionEmail(FormView):
+    """Allows requested reviewers to send an email to requesting editors
+    after they have responded to a peer review request.
+    """
+    template_name = 'shared/editable_notification_email.html'
+    form_class = forms.CustomEmailForm
+
+    @method_decorator(is_reviewer)
+    def dispatch(self, request, *args, **kwargs):
+        self.proposal_id = self.kwargs['proposal_id']
+        self.assignment_id = self.kwargs['assignment_id']
+        self.access_key = self.kwargs.get('access_key')
```

Comment:
Are you using `dict.get()` vs `dict[key]` to distinguish between required and optional `kwargs`?
(just out of interest)

Reply:
Yeah, that's the reasoning behind this.